### ISSUE TITLE
Contributing to this repo requires large downloads

### ISSUE THREAD

**Issue by traviswimer** [chukong/cocos-docs#904]
This repo is surprisingly complex for only containing documentation, and it took a decent amount of time for me to get it working. I think this is unfortunate, because it likely discourages a lot of people from contributing.

One thing I think would be helpful is reducing download size. Here are several areas where it potentially could be fixed:
- The repo is huge ( >500mb )
  *\* It looks like this repo contains all the documentation for all of cocos2d. Would it make sense to break this into separate repos? This is especially true for legacy APIs, since this repo will just continue to grow indefinitely if outdated documentation is never removed.
- LaTex is huge ( ~2.5gb )
  *\* I installed BasicTeX instead, and I think it was ~100mb. Is there any reason not suggest downloading this instead of the full version in the README.md?

**Comment by slackmoehrle** [comment]
Hello,

Separate repos for us have not worked for us in the past.

LaTex is only needed to build the PDF. Perhaps I can add some flags so that you don't need to build everything. I'll add this to my list of tasks.

**Comment by AndyDentFree** [comment]
I worked on the Realm Xamarin SDK team for nearly two years and we went through a similar evolution where a single repo did the full website, including all SDK doc and articles, including translations.

If that becomes very active, it's **really** hard to track contributions - API work being done on _feature branches_ can fall a long way behind the head.

So, hoping for some improvements given the [August 2017](http://discuss.cocos2d-x.org/t/answering-the-questions-about-cocos-creator-the-engine-and-editor/38665) thread about the new directions, I suggest trying to streamline the doc and build process before getting too many contributors on board.

One approach that helped us was having two cloud build pipelines - a _development_ build lets you build the docs and confirm your changes work. The _production_ build had limited access and built the live site. For the most part, that meant we could just checkout the repo and work in simple Markdown.
